skopt/callbacks.py

```diff
@@ -271,6 +271,22 @@ def _criterion(self, result):
             return None
 
 
+class ThresholdStopper(EarlyStopper):
+    """
+    Stop the optimization if the best func_vals is lower than the given
+    threshold
+    """
+    def __init__(self, threshold: float) -> bool:
+        super(EarlyStopper, self).__init__()
+        self.threshold = threshold
+
+    def _criterion(self, result):
+        func_vals = np.sort(result.func_vals)
```

Comment:
The snippet below is O(n) (vs. provided O(nlogn)) complexity:
```py
return np.any(result.func_vals < self.threshold)
```

Reply:
You're right
I had to do 
```python
return np.any([val < self.threshold for val in result.func_vals])
```
to avoid
```
TypeError: '<' not supported between instances of 'list' and 'float'
```

Reply:
`np.min(result.func_vals) < self.threshold` also works.